edelphi/src/main/java/fi/metatavu/edelphi/paytrail/HttpClientIOHandler.java

```diff
@@ -61,8 +73,33 @@ public IOHandlerResult doPost(String merchantId, String merchantSecret, String u
       } finally {
         EntityUtils.consume(entity);
       }
-	  }
+	  } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e) {
+	    logger.log(Level.SEVERE, "Failed to create client", e);
+	    throw new IOException(e);
+    }
 	}
 
+  private CloseableHttpClient createHttpClient(CredentialsProvider credsProvider) throws KeyManagementException, NoSuchAlgorithmException, KeyStoreException {
+    return HttpClients.custom()
+      .setDefaultCredentialsProvider(credsProvider)
+      .setSSLSocketFactory(createSSLConnectionSocketFactory())
+      .build();
+  }
+
+	private SSLConnectionSocketFactory createSSLConnectionSocketFactory() throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+	  SSLContextBuilder builder = new SSLContextBuilder();
+    
+	  builder.loadTrustMaterial(null, new TrustStrategy() {
+
+      @Override
+      public boolean isTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        // TODO: Fix this
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)